### PR TITLE
Add level 1 of native golang Detection code

### PR DESCRIPTION
**Pull request by vadmeste**
Not completed yet, do not review

### CODE REVIEW

**Comment by vadmeste** [comment]
@fwessels, you can now test `samples/face-detect.go` with `samples/data/images/lena.pgm`, it will show correct face coordinates when comparing with the C++ version. Besides, we are now relying on the original Simd library.

This PR not complete yet (grep FIXME)

**Comment by fwessels** [comment]
Nice, cool, will take a look.

File: rectangle.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package gocv
+
+type Rect struct {
```
Comment (fwessels) [inline rectangle.go:3]: Maybe we can use image.Rectangle later (quite some functionality overlap)

File: size.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package gocv
+
+type Size Point
```
Comment (fwessels) [inline size.go:3]: Same, here, maybe use image.Point

**Review by fwessels — APPROVED** [review]

**Pull request merged**